File: lucafontanili.designpatterns/src/main/java/com/lucafontanili/designpatterns/behavioral/state/StateClient.java
package com.lucafontanili.designpatterns.behavioral.state;

import java.io.IOException;

import com.lucafontanili.designpatterns.utilities.AbstractClient;
import com.lucafontanili.designpatterns.utilities.InvalidArgumentException;

public class StateClient extends AbstractClient {

    public static void main(String[] args) throws InvalidArgumentException, IOException {
	main(new StateClient());
    }

    @Override
    public void run() throws InvalidArgumentException, IOException {
	TCPConnection tcpConnection = new TCPConnection();
	tcpConnection.execute(TCPState.CLOSE);
	tcpConnection.execute(TCPState.OPEN);
	tcpConnection.execute(TCPState.CLOSE);
	tcpConnection.execute(TCPState.ESTABLISH);
	tcpConnection.execute(TCPState.CLOSE);
    }

}

File: lucafontanili.designpatterns.python/src/behavioral/state/StateClient.py
'''
Created on Nov 27, 2016

@author: Admin
'''
from TCPConnection import *
from src.behavioral.state import TCPConnection

def main():
    tcp = TCPConnectionClass()
    tcp.execute(TCPConnection.states['close'])
    tcp.execute(TCPConnection.states['open'])
    tcp.execute(TCPConnection.states['close'])
    tcp.execute(TCPConnection.states['establish'])
    tcp.execute(TCPConnection.states['close'])

if __name__ == "__main__" :
    main()